### PR TITLE
Shuttles no longer disable buckling

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl:Crazylemon
fix: Shuttles no longer make buckling useless
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
The question I have is....on TG, this isn't a problem and they have this exact code...any idea why?

Are they just setting the `loc` directly for shuttles, somewhere?

**Comment by Crazylemon64** [comment]
That's exactly it - -tg- uses loc setting in their `onShuttleMove`, while we use `forceMove` in our `onShuttleMove` - @tigercat2000 states this is necessary in order for our client lighting to work. I figured the buckling thing I ported over was a thing that was patched in on -tg- in order to fix weird cases - and outside of admin goofery, I can't really think of anything else that `forceMove`'s a buckled individual far away (besides the shuttle example of course).

**Comment by Fox-McCloud** [comment]
@Crazylemon64 

Teleportation is a major one.

**Comment by Crazylemon64** [comment]
The unbuckle on `forceMove` wasn't a thing before the vision refactor, but...
Hmm. `forceMove` is *very* commonly called, but the timer system is intended to be low-overhead in comparison to `spawn`, so hopefully it'll be up to the task...

**Comment by Krausus** [comment]
I'd bet that force-moving buckled mobs doesn't happen very often. You wouldn't make the timer unless the mob is buckled, of course.

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -148,17 +148,31 @@
...
 
 /mob/living/forceMove(atom/destination)
-	stop_pulling()
 	if(buckled)
-		buckled.unbuckle_mob(src,force=1)
-	// in lieu of "unbuckle_all_mobs"
+		addtimer(src, "checkBuckled", 1, TRUE)
 	if(buckled_mob)
-		unbuckle_mob(buckled_mob,force=1)
+		addtimer(src, "checkBuckledMob", 1, TRUE)
```
Comment (Krausus) [inline code/game/atoms_movable.dm]: No need for a second proc here...
`addtimer(buckled_mob, "checkBuckled", 1, TRUE)`

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -148,17 +148,31 @@
...
+	if(pulling)
+		addtimer(src, "checkPull", 1, TRUE)
 	. = ..()
 	if(client)
 		reset_perspective(destination)
 	update_canmove() //if the mob was asleep inside a container and then got forceMoved out we need to make them fall.
 
+/mob/living/proc/checkBuckled()
+	if(!(buckled in loc))
+		buckled.unbuckle_mob(src, force=1)
```
Comment (Krausus) [inline code/game/atoms_movable.dm]: No sanity check for `buckled` becoming null before the timer fires.

Also, this proc probably belongs with the rest of the buckling code.

File: code/game/atoms_movable.dm [hunk 3]
```diff
@@ -148,17 +148,31 @@
...
+
+/mob/living/proc/checkBuckledMob()
+	// in lieu of "unbuckle_all_mobs"
+	if(!(buckled_mob in loc))
+		unbuckle_mob(buckled_mob, force=1)
+
+/mob/living/proc/checkPull()
+	if(!(pulling in orange(1)))
+		stop_pulling()
```
Comment (Krausus) [inline code/game/atoms_movable.dm]: And this proc probably belongs with the `start`/`stop_pulling` procs.

**Pull request merged**